addons/hw_posbox_homepage/static/src/app/components/dialog/UpdateDialog.js

```diff
@@ -0,0 +1,96 @@
+/* global owl */
+
+import useStore from "../../hooks/useStore.js";
+import { BootstrapDialog } from "./BootstrapDialog.js";
+import { LoadingFullScreen } from "../LoadingFullScreen.js";
+
+const { Component, xml, useState, markup } = owl;
+
+export class UpdateDialog extends Component {
+    static props = {};
+    static components = { BootstrapDialog, LoadingFullScreen };
+
+    setup() {
+        this.store = useStore();
+        this.state = useState({
+            initialization: true,
+            commitHtml: "",
+            loading: false,
+            waitRestart: false,
+            upgradeData: [],
+        });
+    }
+
+    onClose() {
+        this.state.initialization = [];
+    }
+
+    async getUpgradeData() {
+        try {
+            const data = await this.store.rpc({
+                url: "/hw_posbox_homepage/upgrade",
+            });
+
+            this.state.upgradeData = data;
+            this.state.commitHtml = markup(data.commit);
+            this.state.initialization = false;
+        } catch {
+            console.warn("Error while fetching data");
+        }
+    }
+
+    async upgradeIotBox() {
+        console.warn("Not implemented yet");
```

Comment:
@Ysoroko should this be implemented?

Reply:
It's currently a broken feature so while this is a bit ugly it's not wrong